packages/atomic/src/components/recommendations/atomic-recs-list/atomic-recs-list.tsx

```diff
@@ -187,39 +256,141 @@ export class AtomicRecsList implements InitializableComponent<RecsBindings> {
     }
   }
 
-  private get resultListCommonState(): ResultListCommonState<RecsResult> {
-    return {
-      firstSearchExecuted: this.recommendationListState.searchResponseId !== '',
-      isLoading: this.recommendationListState.isLoading,
-      hasError: this.recommendationListState.error !== null,
-      hasResults: this.recommendationListState.recommendations.length !== 0,
-      results: this.recommendationListState.recommendations,
-      searchResponseId: this.recommendationListState.searchResponseId,
-    };
-  }
-
   private renderHeading() {
     if (!this.label) {
       return;
     }
 
-    const shouldHide =
-      this.resultListCommonState.hasError ||
-      (this.resultListCommonState.firstSearchExecuted &&
-        !this.resultListCommonState.hasResults);
+    if (this.resultListCommonState.hasError) {
+      return;
+    }
+
+    if (
+      this.resultListCommonState.firstSearchExecuted &&
+      !this.resultListCommonState.hasResults
+    ) {
+      return;
+    }
 
     return (
-      <Heading
-        level={this.headingLevel}
-        part="label"
-        class={shouldHide ? 'hidden' : ''}
-      >
+      <Heading level={this.headingLevel} part="label" class="m-0 mb-2">
         {this.bindings.i18n.t(this.label)}
       </Heading>
     );
   }
 
+  private get currentIndex() {
+    return Math.abs(
+      (this.currentPage * this.numberOfRecommendations) %
+        this.recommendationListState.recommendations.length
+    );
+  }
+
+  private get subsetRecommendations() {
+    return this.recommendationListState.recommendations.slice(
+      this.currentIndex,
+      this.currentIndex + this.numberOfRecommendations
+    );
+  }
```

Comment:
I think it would be nicer if some of the carousel/pagination logic was extracted into a functional component.